### PR TITLE
Test guava upgrade

### PR DESCRIPTION
**Pull request by timja**
https://github.com/jenkinsci/jenkins/pull/5059

### CODE REVIEW

**Comment by timja** [comment]
@basil would you be able to handle the timestamper part of this? It's in the test code and I don't really understand how to adapt the test

`Ranges` and `DiscreteDomains` are removed, replaced by `Range` and `DiscreteDomain`, there's no `asSet` method though

https://github.com/jenkinsci/timestamper-plugin/blob/5d434ae9a1e9c5c5cfacbb4202b62bb1a2b02acf/src/test/java/hudson/plugins/timestamper/TimestamperApiTestUtil.java#L91

**Comment by basil** [comment]
@timja I fixed Timestamper in jenkinsci/timestamper-plugin#108.

**Comment by timja** [comment]
Seems like workflow-basic-steps is reliably failing

File: bom-latest/pom.xml [hunk 1]
```diff
@@ -106,7 +106,7 @@
             <dependency>
                 <groupId>org.jenkins-ci.plugins.workflow</groupId>
                 <artifactId>workflow-basic-steps</artifactId>
-                <version>2.23</version>
+                <version>2.24-rc866.001647014233</version> <!-- TODO https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/147 -->
```
Comment (timja) [inline bom-latest/pom.xml:109]:
```suggestion
                <version>2.24-rc868.f16f213ddbb3</version> <!-- TODO https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/147 -->
```

File: Jenkinsfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 def buildNumber = BUILD_NUMBER as int; if (buildNumber > 1) milestone(buildNumber - 1); milestone(buildNumber) // JENKINS-43353 / JENKINS-58625
 
 def mavenEnv(Map params = [:], Closure body) {
-    node('maven') { // no Dockerized tests; https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#container-agents
+    node('maven && kubernetes') { // no Dockerized tests; https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#container-agents
```
Comment (timja) [inline Jenkinsfile:4]:
```suggestion
    node('maven') { // no Dockerized tests; https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#container-agents
```

File: bom-latest/pom.xml [hunk 3]
```diff
@@ -106,7 +106,7 @@
             <dependency>
                 <groupId>org.jenkins-ci.plugins.workflow</groupId>
                 <artifactId>workflow-basic-steps</artifactId>
-                <version>2.23</version>
+                <version>2.24-rc868.f16f213ddbb3</version> <!-- TODO https://github.com/jenkinsci/workflow-basic-steps-plugin/pull/147 -->
```
Comment (timja) [inline bom-latest/pom.xml:109]:
```suggestion
                <version>2.24</version>
```

**Pull request closed without merging**